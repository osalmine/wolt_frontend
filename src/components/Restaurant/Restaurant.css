.rounded-corners canvas {
	border-radius: 8px;
}

.dot {
	height: 10px;
	width: 10px;
	border-radius: 50%;
	display: inline-block;
	margin-bottom: 1px;
}

.dot.offline {
	background-color: red;
}

.dot.online {
	background-color: rgb(0, 172, 0);
}

.restaurant-wrapper {
	position: relative;
	width: 65%;
	height: 90%;
	min-height: 9rem;
	margin: 20px auto 0;
	/* border: 1px solid black; */
	/* padding: 10px; */
}

.text-container h2 {
	/* float: left; */
	/* position: absolute;
	bottom: 0;
	left: 0; */
	margin: 0;
	color: #282c34;
	font-size: 2rem;
	max-width: 75%;
	/* margin: 3rem; */
	/* position: absolute;
	left: 50%;
	transform: translate(-50%, -50%); */
	z-index: 1;
}

.text-container p {
	/* float: right; */
	/* position: absolute;
	bottom: 0;
	right: 0; */
	/* margin: 3rem; */
	color: #282c34;
	font-size: 20px;
	/* position: absolute;
	left: 50%;
	transform: translate(-50%, -50%);
	top: 9%; */
	z-index: 1;
	margin: 0;
}

.text-container {
	position: absolute;
	/* margin: 3rem; */
	/* margin: 0 0 3rem 0; */
	/* bottom: 0; */
	margin-top: 20px;
	display: flex;
	/* justify-content: center; */
	justify-content: space-between;
	align-items: center;
	height: 20px;
	/* width: calc(100% - 1rem); */
	width: 100%;
	/* border: 1px solid green; */
	/* padding: 0 1rem 0 0; */
}

@media (max-width: 576px) {
	.text-container {
		/* margin-top: 10px; */
		bottom: 12px;
		height: auto;
	}
	.text-container p {
		font-size: 15px;
		align-self: flex-end;
	}
	.text-container h2 {
		font-size: 1rem;
	}
	.restaurant-wrapper {
		margin-bottom: 10px;
		width: 70%;
		height: 95%;
	}
}