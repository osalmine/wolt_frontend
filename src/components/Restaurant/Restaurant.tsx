import React from 'react';
import { Restaurant } from '../RestaurantSection/RestaurantSection';
import { Blurhash } from "react-blurhash";
import './Restaurant.css';

interface Props {
	restaurant: Restaurant;
	displayBlurhash?: boolean;
}

const RestaurantComponent: React.FC<Props> = ({ restaurant, displayBlurhash = true }) => {
	return (
		<div className="restaurant-wrapper">
			{displayBlurhash && 
			<Blurhash 
				hash={restaurant.blurhash}
				width={'100%'}
				height={'80%'}
				className="rounded-corners d-block"
			/>}
			<div className="text-container">
				<h2>{restaurant.name}</h2>
				<p>{restaurant.online ? "online" : "offline"} <span className={restaurant.online ? "dot online" : "dot offline"} /></p>
			</div>
		</div>);
};

export default RestaurantComponent;