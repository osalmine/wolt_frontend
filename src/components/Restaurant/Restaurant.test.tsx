import Restaurant from './Restaurant';
import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

const restaurantObject = {
	"blurhash": "UAN=8k?LS~M:ErJFs%t0MDMWRqo@%BxSV{RX",
	"launch_date": "2020-04-20",
	"location": [
		24.938082,
		60.17626
	],
	"name": "Sea Chain",
	"online": true,
	"popularity": 0.956990414084132
};

describe('Restaurant component tests', () => {
	test('has name from inputJson', () => {
		const component = render(
			<Restaurant restaurant={restaurantObject} displayBlurhash={false} />
		);
	
		expect(component.container).toHaveTextContent('Sea Chain');
	});
	
	test('has oneline status from inputJson', () => {
		const component = render(
			<Restaurant restaurant={restaurantObject} displayBlurhash={false} />
		);
	
		expect(component.container).toHaveTextContent('online');
	});
	
	test('matches snapshot', () => {
		const tree = renderer.create(
			<Restaurant restaurant={restaurantObject} displayBlurhash={false} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});