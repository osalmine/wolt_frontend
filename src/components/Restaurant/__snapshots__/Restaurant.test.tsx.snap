// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant component tests matches snapshot 1`] = `
<div
  className="restaurant-wrapper"
>
  <div
    className="text-container"
  >
    <h2>
      Sea Chain
    </h2>
    <p>
      online
       
      <span
        className="dot online"
      />
    </p>
  </div>
</div>
`;
